### PR TITLE
[AKS] az aks use-dev-spaces: Adding endpoint type option to use-dev-spaces command to customize the endpoint created on a controller

### PR DESCRIPTION
**Pull request by rakeshvanga**
**History Notes:**  
[ACS] az aks use-dev-spaces: Adding endpoint type option to the use-dev-spaces command to customize the endpoint created on an Azure Dev Spaces controller.
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S165.

File: src/azure-cli/azure/cli/command_modules/acs/custom.py [hunk 1]
```diff
@@ -2120,7 +2120,7 @@ def aks_upgrade(cmd, client, resource_group_name, name, kubernetes_version, cont
 DEV_SPACES_EXTENSION_MODULE = 'azext_dev_spaces.custom'
 
 
-def aks_use_dev_spaces(cmd, client, name, resource_group_name, update=False, space_name=None, prompt=False):
+def aks_use_dev_spaces(cmd, client, name, resource_group_name, update=False, space_name=None, endpoint_type=None, prompt=False):
```
Comment (amsoedal) [inline src/azure-cli/azure/cli/command_modules/acs/custom.py]: endpoint_type=Public ?

Comment (amsoedal) [reply src/azure-cli/azure/cli/command_modules/acs/custom.py]: Even though I see we are setting default to Public in _params.py

Comment (amsoedal) [inline src/azure-cli/azure/cli/command_modules/acs/custom.py]: let's be explicit about the default we are expecting: endpoint_type=Public

**Comment by sanar-microsoft** [comment]
@qwordy Can you please trigger the build pipeline again? Thanks

**Comment by rakeshvanga** [comment]
@qwordy The test cases failing are not related to the changes in this PR. So can you re-run the pipeline?

**Comment by rakeshvanga** [comment]
@qwordy Thanks for re-running the pipeline.
The pipeline failed with following error:
```Pipeline configuration error:
Job CredScan: Step task reference is invalid. The task name CredScan is ambiguous. Specify one of the following identifiers to resolve the ambiguity: ea576cd4-c61f-48f8-97e7-a3cb07b90a6f, e587fa4a-fc98-44fd-ab3e-ddaa3025219a
Job CredScan: Step task reference is invalid. The task name PostAnalysis is ambiguous. Specify one of the following identifiers to resolve the ambiguity: f5679091-e6da-4974-a8dc-0eec03a8ea63, 18ff34ab-7385-4b57-b5d5-a0fcc7b5b274```

Can this PR be merged by-passing this check? Or can you help resolve this pipeline issue?

**Comment by qwordy** [comment]
> 
> 
> @qwordy Thanks for re-running the pipeline.
> The pipeline failed with following error:
> 
> ```
> Pipeline configuration error:
> Job CredScan: Step task reference is invalid. The task name CredScan is ambiguous. Specify one of the following identifiers to resolve the ambiguity: ea576cd4-c61f-48f8-97e7-a3cb07b90a6f, e587fa4a-fc98-44fd-ab3e-ddaa3025219a
> Job CredScan: Step task reference is invalid. The task name PostAnalysis is ambiguous. Specify one of the following identifiers to resolve the ambiguity: f5679091-e6da-4974-a8dc-0eec03a8ea63, 18ff34ab-7385-4b57-b5d5-a0fcc7b5b274
> ```
> 
> Can this PR be merged by-passing this check? Or can you help resolve this pipeline issue?

@haroldrandom @fengzhou-msft Can you help with this problem? Thanks a lot!

**Comment by haroldrandom** [comment]
@rakeshvanga your branch is way too behind dev branch, could you please rebase/merge your branch against the last commit of dev?
I guess it's causing something weird in ADO such that we can read the detail of this build

**Comment by qwordy** [comment]
> 
> 
> @rakeshvanga your branch is way too behind dev branch, could you please rebase/merge your branch against the last commit of dev?
> I guess it's causing something weird in ADO such that we cannot read the detailed output of this build :(

Do you mean CI will use pipeline definition in PR instead of dev branch?

**Comment by rakeshvanga** [comment]
> @rakeshvanga your branch is way too behind dev branch, could you please rebase/merge your branch against the last commit of dev?
> I guess it's causing something weird in ADO such that we cannot read the detailed output of this build :(

Sure I have updated my branch and pushed new changes

**Comment by haroldrandom** [comment]
> > @rakeshvanga your branch is way too behind dev branch, could you please rebase/merge your branch against the last commit of dev?
> > I guess it's causing something weird in ADO such that we cannot read the detailed output of this build :(
> 
> Sure I have updated my branch and pushed new changes

CI is fine. Good to go. @qwordy

**Pull request merged**